ssh-server/main.go

```
@@ -0,0 +1,137 @@
+	}
+	log.Printf("Listening on %s...", serverAddress)
+	return config, listener, nil
+}
+
+func getConfiguration() (*ssh.ServerConfig, error) {
+	config := &ssh.ServerConfig{NoClientAuth: true}
+
+	privateBytes, err := ioutil.ReadFile("id_rsa")
+	if err != nil {
+		return nil, fmt.Errorf("reading private server key: %v", err)
+	}
+	private, err := ssh.ParsePrivateKey(privateBytes)
+	if err != nil {
+		return nil, fmt.Errorf("parsing private server key: %v", err)
+	}
+	config.AddHostKey(private)
+	return config, nil
+}
+
+func handleConnection(conn net.Conn, serverConfig *ssh.ServerConfig) error {
+	_, chans, reqs, err := ssh.NewServerConn(conn, serverConfig)
+	if err != nil {
+		return fmt.Errorf("handshake: %v", err)
+	}
+
+	// Discard out-of-band SSH requests (not supported by this server).
+	go ssh.DiscardRequests(reqs)
+
+	for newChannel := range chans {
+		if err := serviceChannel(newChannel); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func serviceChannel(newChannel ssh.NewChannel) error {
+	if newChannel.ChannelType() != "session" {
+		newChannel.Reject(ssh.UnknownChannelType, "unknown channel type")
+		return nil
+	}
+	channel, requests, err := newChannel.Accept()
+	if err != nil {
+		return fmt.Errorf("accepting channel: %v", err)
+	}
+	defer channel.Close()
+
+	// Start the command with a pseudo-terminal.
+	bashPTY, err := pty.Start(exec.Command("bash"))
+	if err != nil {
+		return fmt.Errorf("starting pty: %v", err)
+	}
+	defer bashPTY.Close()
+
+	go func(in <-chan *ssh.Request) {
+		for req := range in {
+			// Tell the client we accept pty and shell commands
+			req.Reply(req.Type == "pty-req" || req.Type == "shell", nil)
+
+			var winSize struct {
+				Width  uint32
+				Height uint32
+			}
+
+			if req.Type == "pty-req" {
+				skip := binary.BigEndian.Uint32(req.Payload)
+				buf := bytes.NewReader(req.Payload[4+skip:])
+				err = binary.Read(buf, binary.BigEndian, &winSize)
```

This doesn't do anything...  presumably you should either completely ignore the error (questionable?) or reject the message via the SSH protocol